cmd/policy_group_ls.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+	"github.com/spf13/cobra"
+)
+
+func newPolicyGroupCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "group",
+		Short: "Manage policy groups",
+		Args:  cmdutil.NoArgs,
+	}
+
+	cmd.AddCommand(newPolicyGroupLsCmd())
+	return cmd
+}
+
+func newPolicyGroupLsCmd() *cobra.Command {
+	var jsonOut bool
+	var cmd = &cobra.Command{
+		Use:   "ls [org-name]",
+		Args:  cmdutil.MaximumNArgs(1),
+		Short: "List all Policy Groups for a Pulumi organization",
+		Long:  "List all Policy Groups for a Pulumi organization",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, cliArgs []string) error {
+			// Get backend.
+			b, err := currentBackend(display.Options{Color: cmdutil.GetGlobalColorization()})
+			if err != nil {
+				return err
+			}
+
+			// Get organization.
+			var orgName string
```

Comment:
The `filestate` backend doesn't have any notion of organization... this code is fine, since `ListPolicyGroups` will fail later (because the `filestate` backend doesn't support policy groups, either). But this hints that maybe we need to first check that the backend has type `httpstate.Backend` and/or check that the provided backend implements the `backend.PolicyAsCode` interface. That would help us cut down on the duplication and "noops" between the two backends. e.g.

```go
// PolicyAsCode is an interface implemented by any backend that supports the policy-as-code feature.
// As of now, that is just the `httpstate` backend.
type PolicyAsCode interface {
    ListPolicyGroups(...)
    ...
```

Not something to address with this PR, just thinking aloud. IIRC the platform team was going to look into refactoring this code to be easier to maintain this milestone. Perhaps this is something that could help in that effort.